subprojects/docs/src/samples/userguide/files/fileTrees/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,56 @@
+import org.gradle.internal.impldep.jcifs.Config
+
+// tag::define[]
+// Create a file tree with a base directory
+var tree: ConfigurableFileTree = fileTree("src/main")
+
+// Add include and exclude patterns to the tree
+tree.include("**/*.java")
+tree.exclude("**/Abstract*")
+
+// Create a tree using path
+tree = fileTree("src").include("**/*.java") as ConfigurableFileTree
+
+// Create a tree using closure
+tree = fileTree("src") {
+    include("**/*.java")
+}
+
+// Create a tree using a map
+tree = fileTree(mapOf("dir" to "src", "include" to "**/*.java"))
+tree = fileTree(mapOf("dir" to "src", "includes" to listOf("**/*.java", "**/*.xml")))
+tree = fileTree(mapOf("dir" to "src", "include" to "**/*.java", "exclude" to "**/*test*/**"))
```

Comment:
:x: these three can be simplified using the `vararg args: Pair<String, Any>` taking overloads:
```kotlin
tree = fileTree("dir" to "src", "include" to "**/*.java")
tree = fileTree("dir" to "src", "includes" to listOf("**/*.java", "**/*.xml"))
tree = fileTree("dir" to "src", "include" to "**/*.java", "exclude" to "**/*test*/**")
```